### DISCUSSION TITLE
Change default IP address AP mode

### DISCUSSION THREAD

**Discussion by nastiliano** [micropython/micropython#12915] [General]
Hello, everyone.

Is not explained clearly how to change the default IP address (192.168.4.1) in AP mode in the Micropython manual. I know is possible but putting this code doesn't work:

    wlan = network.WLAN(network.AP_IF)
    wlan.ifconfig("192.168.1.1") 
    ## I tried (  wlan.ifconfig(("192.168.1.1" , "" , "" , ""))  and wlan.ifconfig(("192.168.1.1" , "255.255.255.0" , "192.168.1.1" , "192.168.1.1"))  )
    wlan.config(max_clients = 1 , essid = self.ssid , authmode = 3 , password = self.password)
    wlan.active(True)

Otherwise in the STATION mode all works:

    wlan = network.WLAN(network.STA_IF)
    wlan.active(True)
    wlan.ifconfig((self._station_IP , self.subnet_mask , self.gateway_IP , self.gateway_IP))
    wlan.config(dhcp_hostname = self.host)
    wlan.connect(self.ssid , self.password)

Thank you.

**Comment by sophiedegran** [comment]
The default IP adress in AP mode seem to be hard define in dhcpserver.c (192.168.4.1).. Try to change it and compile it should work

**Reply by nastiliano** [reply]
Hello.

I read in some ESP32 forums that in "C" language is possible, the question is if we can do the same with micropython.


Thank you.

**Reply by sophiedegran** [reply]
Yes change in dhcpserver.c line 68
# define DEFAULT_DNS MAKE _IP4 (192.168.4.1)
to what ip adress you want.. (give a private ip adress) for example:
10.0.0.1 .. 192.168.1.1 
re-complil micropython and it should work ☺

**Reply by nastiliano** [reply]
Yes, I understand you, but I want to change the IP address "on_the_go".

I have a PC application and an "user" changes the configuration and the devices re-starts with its "WiFi" configuration.

This solution is not valid for me

Thank you.

**Reply by sophiedegran** [reply]
Sorry you have to write something that close dhcpserver / affect a new ip adress / re-start it and fortunally re-start the ap mode ... It's not complicate but i don't understand why you want to change 'on the go' ip ap lol ☺

**Reply by nastiliano** [reply]
Mmmm... sorry I think I didn't explain me correctly.

I want to START my ESP32 in AP mode but with a different IP, not the preset IP (192.168.4.1).

I want to START with, for example, 192.168.1.1, and then restart the device with 192.168.2.1... etc.

My PC app "configure" my ESP32 device and restart it.

But all in micropython, not in C or recompile or this kind of things...

Thank you very much.

PD: In "C" is very simple:
    IPAddress IP = WiFi.softAPIP();

**Reply by sophiedegran** [reply]
When i have time i would wrote something in C that you can call in micropython.. it's not complicate

**Reply by slmnv5** [reply]
it is plain misleading to provide API for WIFI access point where you can specify IP address, but then always use the same hardcoded IP. :( More so if it is not complicate to fix as the last message states

**Reply by nastiliano** [reply]
Thank you for your response

This isssue is solved long time ago

Best regards

**Reply by ericty91** [reply]
hi all. can you post your solution please? i encountered the same issue here with a esp-32 board and micropython 1.23 firmware

**Reply by nastiliano** [reply]
The solution was, there's no solution...

I coded my program assuming that... sorry
